pkg/api/security/cert/cert_getter_dca.go

```diff
@@ -0,0 +1,136 @@
+	if clusterCAPath == "" || clusterCAKeyPath == "" {
+		return &clusterCAData{
+			enableTLSVerification: enableTLSVerification,
+		}, nil
+	}
+
+	// Read cluster CA certificate and private key from disk
+	caCert, caPrivKey, err := ReadClusterCA(clusterCAPath, clusterCAKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read cluster CA cert and key: %w", err)
+	}
+
+	return &clusterCAData{
+		enableTLSVerification: enableTLSVerification,
+		caCert:                caCert,
+		caPrivKey:             caPrivKey,
+	}, nil
+}
+
+// buildClusterClientTLSConfig creates the TLS configuration for cluster client communication
+// using pre-read cluster CA data
+func (c *clusterCAData) buildClusterClientTLSConfig() (*tls.Config, error) {
+	// Default to insecure configuration
+	clusterClientConfig := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	// If TLS verification is enabled, configure proper certificate validation
+	if c.enableTLSVerification {
+		// It's not possible to have TLS verification enabled without a CA certificate
+		if c.caCert == nil || c.caPrivKey == nil {
+			return nil, fmt.Errorf("cluster_trust_chain.enable_tls_verification cannot be true if cluster_trust_chain.ca_cert_file_path or cluster_trust_chain.ca_key_file_path is not set")
+		}
+
+		clusterClientCertPool := x509.NewCertPool()
+		clusterClientCertPool.AddCert(c.caCert)
+		clusterClientConfig = &tls.Config{
+			RootCAs: clusterClientCertPool,
+		}
+	}
+
+	return clusterClientConfig, nil
+}
+
+// setupCertificateFactoryWithClusterCA configures the certificate factory with cluster CA
+// and determines additional SANs based on the agent flavor and configuration
+func (c *clusterCAData) setupCertificateFactoryWithClusterCA(config configModel.Reader, factory *certificateFactory) error {
+	// Only proceed if cluster CA data is available
+	if c.caCert == nil || c.caPrivKey == nil {
+		return nil
+	}
+
+	factory.caCert = c.caCert
+	factory.caPrivKey = c.caPrivKey
+
+	isCLC := config.GetBool("clc_runner_enabled")
+	isClusterAgent := flavor.GetFlavor() == flavor.ClusterAgent
+	isOther := !isCLC && !isClusterAgent
+
+	// If the process is not a CLC Runner or Cluster Agent, we don't need to add any SANs
+	if isOther {
+		return nil
+	}
+
+	var serverHost string
+
+	// If the process is a Cluster Agent, add the external IP and DNS name to the SANs
+	if isClusterAgent {
+		clusterAgentEndpoint, err := configutils.GetClusterAgentEndpoint()
+		if err != nil {
+			return fmt.Errorf("unable to get cluster agent endpoint: %w", err)
+		}
+		parsedURL, err := url.Parse(clusterAgentEndpoint)
+		if err != nil {
+			return fmt.Errorf("unable to parse cluster agent endpoint URL: %w", err)
+		}
+
+		serverHost, _, err = net.SplitHostPort(parsedURL.Host)
+		if err != nil {
+			return fmt.Errorf("unable to get pod IP from cluster agent endpoint: %w", err)
+		}
+	}
+
+	// If the process is a CLC Runner, add the CLC Runner host to the SANs
+	if isCLC {
```

Comment:
Can the cluster agent be the CLC runner as well? I imagine yes, but just double-checking for my own learning 😄